### PR TITLE
User specified number of significant figures for numerical outputs

### LINKED ISSUE #282: User specified number of significant figures for numerical result outputs
* Similar to #272
* Allow user to provide `round_to` kwarg to the `summary` method. That will in turn involve changes to the `print_coefficients` method

### PR DESCRIPTION
**Pull request by drbenvincent**
* Closes #282 
* Doctests and tests pass locally.
* Docstrings updated to reflect the new output you get.

### CODE REVIEW

**Comment by drbenvincent** [comment]
Pfft. These doctests are brittle.

**Comment by jpreszler** [comment]
There are definitely some that are particularly problematic, and will continue to be so without some significant changes. I think the value comes over the long-term, but it will be hard to get there without some improvements. The options I see:
1. turn them off, at least temporarily.
2. Split the example and output into separate docstrings (could be difficult for some longer examples). This would flag the biggest problems (an exception from input changes) but would make the actual example output static. This could be a good temporary fix in some places.
3. Replace the more problematic examples. I generally grabbed things from the other docs which are nice realistic example but have coefficients near 0 or HDI's near or including 0. This is perhaps the biggest issue since the output may change sign in a way the makes even the pytest numeric accuracy checks too strict. It's likely possible to have examples that show the right output, but sacrifice realism for basic numeric stability that make the doctests far more reliable. 

My vote would be to temporarily turn off the tests and replace the examples for at least PrePostFit (flakey), RegressionDiscontinuity (flakey) and the Instrumented Variables (slow) and turn doctests back on. I should be able to work on this soon, perhaps even getting better examples over the next week.

**Comment by jpreszler** [comment]
created https://github.com/pymc-labs/CausalPy/issues/284. For now, I think you can just comment out the doctest call in the CI untill that issue is resolved.

**Comment by drbenvincent** [comment]
Note to self: I'll finish and merge this PR after #286 is merged

**Comment by drbenvincent** [comment]
Well, I guess I have to increase test coverage

File: causalpy/pymc_experiments.py
Comment (juanitorduz) [inline causalpy/pymc_experiments.py:67]: None (upper case)

File: causalpy/pymc_experiments.py
Comment (juanitorduz) [inline causalpy/pymc_experiments.py:1387]: None (upper case)

**Review by juanitorduz — CHANGES_REQUESTED** [review]
Minor change request ;) Otherwise LGTM.

File: causalpy/pymc_experiments.py
Comment (drbenvincent) [reply causalpy/pymc_experiments.py]: Done: changed this for all instances

**Pull request merged**
